### PR TITLE
fix loss scaling only when compute_loss_func is used

### PR DESCRIPTION
**Pull request by BlackNoodle**
# What does this PR do?

In [#34198](https://github.com/huggingface/transformers/commit/6ba31a8a94bf7cfeaf59ffc3bc9e0b0cd3e25788#diff-ed55888e6665791fe92cc8fc0c499da54f4ace6738551cd9a2591881cda076deR3629), the line `loss *= self.args.gradient_accumulation_steps` was introduced due to `Negate accelerate grad accum div`. This change was made to correct errors encountered during gradient accumulation. However, the scaling should only occur when compute_loss_func is used, so the code was modified accordingly.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@muellerzr 
@ArthurZucker

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
cc @muellerzr `compute_loss_func` is never None right?

**Review by muellerzr — COMMENTED** [review]
No, it should always be used, period. The correct check is if we passed in `num_items_in_batch` or not. If not, then we can mult.

(There are model loss functions which accept `num_items_in_batch` as part of their `forward`)

**Comment by muellerzr** [comment]
This will conflict with https://github.com/huggingface/transformers/pull/33673, however the correct version of this is either if `compute_loss_func` is used **or** `self.model_accepts_loss_kwargs is True`
